File: validSudoku.py
class Solution:
    # @param board, a 9x9 2D array
    # @return a boolean
    def isValidSudoku(self, board):
    	for i in range(9):
    		for j in range(9):
    			print board[i][j],
    		print 

        for i in range(9):
        	t = set()
        	for char in board[i]:
        		if char == '.':
        			continue
        		elif char in t:
        			return False
        		else:
        			t.add(char)

        for j in range(9):
        	t = set()
        	for i in range(9):
        		char = board[i][j]
        		if char == '.':
        			continue
        		elif char in t:
        			return False
        		else:
        			t.add(char)

        print 
        for i in range(3):
        	print '#', i
        	for j in range(3):

        		t = set()
        		for k in range(3):
        			for m in range(3):
        				print board[3*i + k][3*j + m],
        				char = board[3*i + k][3*j + m]
		        		if char == '.':
		        			continue
		        		elif char in t:
		        			return False
		        		else:
		        			t.add(char)
		        print 
		        print i,j
	return True

board = [\
"..5.....6",\
"....14...",\
".........",\
".....92..",\
"5....2...",\
".......3.",\
"...54....",\
"3.....42.",\
"...27.6.."\
]
board = ["......5..",".........",".........","93..2.4..","..7...3..",".........","...34....",".....3...",".....52.."]
s = Solution()
print s.isValidSudoku(board)

File: JAVA/validSudoku.java
public class Solution {
    public boolean isValidSudoku(char[][] board) {
        int probe = 0, temp;
        for (int i=0; i < 9; i++){
            probe = 0;
            for (int j = 0; j < 9; j++){
                if (board[i][j] == '.')
                    continue;
                temp = probe & 1 << (board[i][j] - '1');
                if (temp != 0)
                    return false;
                probe |= 1 << (board[i][j] - '1');
            }
        }        
        for (int i=0; i < 9; i++){
            probe = 0;
            for (int j = 0; j < 9; j++){
                if (board[j][i] == '.')
                    continue;
                temp = probe & 1 << (board[j][i] - '1');
                if (temp != 0)
                    return false;
                probe |= 1 << (board[j][i] - '1');
            }
        }        
        for (int i=0; i < 3; i++){
            for (int j = 0; j < 3; j++){
                probe = 0;
                for (int k = 0; k < 3; k++){
                    for (int q = 0; q < 3; q++){
                        if (board[3*i+k][3*j+q] == '.')
                            continue;
                        temp =  probe & 1 << (board[3*i+k][3*j+q] - '1');
                        if (temp != 0)
                            return false;
                        probe |= 1 << (board[3*i+k][3*j+q] - '1');
                    }
                }
            }
        }
        return true;
    }
}